src/libudev/libudev-device.c

```diff
@@ -156,11 +148,10 @@ _public_ const char *udev_device_get_subsystem(struct udev_device *udev_device)
         assert_return_errno(udev_device, NULL, EINVAL);
 
         r = sd_device_get_subsystem(udev_device->device, &subsystem);
-        if (r < 0) {
-                errno = -r;
-                return NULL;
-        } else if (!subsystem)
-                errno = ENODATA;
+        if (r < 0)
+                return_with_errno(NULL, r);
+        if (!subsystem)
```

Comment:
`sd_device_get_subsystem` returns `-ENOENT` if subsystem is not set, so this check can be dropped.